### PR TITLE
Validate Programming Exercise Test Case Settings

### LINKED ISSUE #3226: Don't Allow Negative Programming Exercise Grading Settings
### Describe the bug
In the programming exercise grading page, we can set the multipliers and similar values to negative numbers which defeats the purpose of a reward when passing a test case. This can also lead to negative scores, which in itself might be a different issue.

#### To Reproduce
0. *You are instructor of a course with one programming exercise*
1. Go to the grading page of the exercise
2. Enter large negative multiplier values for some tests
3. Click save
4. Artemis accepts those negative values
5. Re-evaluate/re-build the solution
6. The solution gets a negative score (if the numbers are negative enough).

#### Expected behavior
- The client does not allow saving negative values and displays an error (maybe via server response).
- The server should reject negative values.

#### Screenshots
- **Settings:**
  image
- **Result:**
  image

### Environment

<details><pre>
 - OS: Win 10
 - Browser: Chromium
 - Artemis-Version: 4.11.4+
</pre></details>

### PR DESCRIPTION
**Pull request by MaisiKoleni**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context
This fixes #3226 and prevents negative test case weights/multipliers/points.

### Description
- Added method to the `ProgrammingExerciseTestCaseService` to check test cases for validity when updates are performed.
- Added translations for the bad request alerts for the client.

### Steps for Testing
0. *You are instructor of a course with one programming exercise*
1. Go to the grading page of the exercise
2. Enter large negative weights/multipliers/points values for some tests
3. Click save
4. Artemis shows you an error mentioning the test case with negative settings.

### Screenshots
image

### CODE REVIEW

File: src/main/webapp/i18n/de/programmingExercise.json [hunk 1]
```diff
@@ -142,7 +142,9 @@
                     "updated": "Die Testfälle wurden erfolgreich gespeichert.",
                     "couldNotBeUpdated": "Die folgenden Testfälle konnten nicht gespeichert werden: {{testCases}}.",
                     "weightSumError": "Die Summe aller Testfall-Gewichte beträgt 0 oder weniger. Bitte stelle mindestens ein Testfall-Gewicht größer als 0 ein.",
-                    "showInactive": "Inaktive Testfälle anzeigen"
+                    "showInactive": "Inaktive Testfälle anzeigen",
+                    "settingNull": "Der Testfall {{testCase}} darf keine Einstellungen haben, die null sind.",
```
Comment (b-fein) [inline src/main/webapp/i18n/de/programmingExercise.json:146]:
```suggestion
                    "settingNull": "Der Testfall {{testCase}} darf keine leeren Einstellungen haben.",
```
I think having `null` in the user-facing error message is problematic, as it results in an easy confusion between the meanings as nullpointer and zero, the latter being allowed.

File: src/main/webapp/i18n/de/programmingExercise.json [hunk 2]
```diff
@@ -142,7 +142,9 @@
...
+                    "settingNegative": "Der Testfall {{testCase}} darf keine Einstellungen mit negativen Zahlen haben."
```
Comment (b-fein) [inline src/main/webapp/i18n/de/programmingExercise.json:147]:
```suggestion
                    "settingNegative": "Der Testfall {{testCase}} darf keine Einstellungen mit negativen Werten haben."
```
(Personal preference, feel free to ignore this if you want to keep the existing one.)

File: src/main/webapp/i18n/en/programmingExercise.json [hunk 3]
```diff
@@ -142,7 +142,9 @@
                     "updated": "The test cases were updated successfully.",
                     "couldNotBeUpdated": "The following test cases could not be updated: {{testCases}}.",
                     "weightSumError": "The sum of all test case weights is 0 or below. Please adjust at least one test case weight to be greater than 0",
-                    "showInactive": "Show inactive test cases"
+                    "showInactive": "Show inactive test cases",
+                    "settingNull": "Test case {{testCase}} must not have settings that are null.",
```
Comment (b-fein) [inline src/main/webapp/i18n/en/programmingExercise.json:146]:
```suggestion
                    "settingNull": "Test case {{testCase}} must not have settings that are empty.",
```
As in the German translation.

File: src/main/webapp/i18n/en/programmingExercise.json [hunk 4]
```diff
@@ -142,7 +142,9 @@
...
+                    "settingNegative": "Test case {{testCase}} must not have settings set to negative numbers."
```
Comment (b-fein) [inline src/main/webapp/i18n/en/programmingExercise.json:147]:
```suggestion
                    "settingNegative": "Test case {{testCase}} must not have settings set to negative values."
```
As in the German translation.

**Review by b-fein — COMMENTED** [review]
Did a code review. Looks good, I only have got a remark about potentially confusing translation strings.

**Review by krusche — APPROVED** [review]
code looks good 👍

**Pull request merged**